codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2HeadersDecoder.java

```diff
@@ -47,17 +43,19 @@ public DefaultHttp2HeadersDecoder() {
     }
 
     public DefaultHttp2HeadersDecoder(boolean validateHeaders) {
-        this(DEFAULT_MAX_HEADER_SIZE, DEFAULT_HEADER_TABLE_SIZE, validateHeaders, 32);
+        try {
+            decoder = new Decoder(DEFAULT_HEADER_LIST_SIZE, DEFAULT_HEADER_TABLE_SIZE, 32);
+        } catch (Http2Exception e) {
+            throw new Error("initial values are invalid!", e);
```

Comment:
Maybe we should just make a default constructor for `Decoder` that doesn't throw?